packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts

```diff
@@ -449,6 +450,17 @@ export class ZoweDatasetNode extends ZoweTreeNode implements IZoweDatasetTreeNod
         return this.etag;
     }
 
+    /**
+     * Returns the imperative.IProfileLoaded profile for this node
+     *
+     * @returns {imperative.IProfileLoaded}
+     */
+    public getProfile(): imperative.IProfileLoaded {
+        const prof = this.profile ?? this.getParent()?.getProfile();
+
+        return Profiles.getInstance().loadNamedProfile(prof.name); // this returns the profile with newer token
```

Comment:
While its highly unlikely, a data set node can technically be created without a profile, so `prof` may be `undefined` for certain nodes. In addition, the `name` property on a profile is optional. Can you please add a check to see if `prof.name` exists before passing it to `Profiles.loadNamedProfile`? This would also avoid any warnings or flags by security scanning tools for the use of `prof.name`.

```suggestion
        // return the latest object for the given profile name
        return prof.name ? Profiles.getInstance().loadNamedProfile(prof.name) : undefined;
```